userspace/libsinsp/sinsp_filtercheck_fspath.cpp

```diff
@@ -355,10 +355,99 @@ uint8_t* sinsp_filter_check_fspath::extract_single(sinsp_evt* evt, uint32_t* len
 
 		if(!std::filesystem::path(m_tstr).is_absolute())
 		{
-			m_tstr = sinsp_utils::concatenate_paths(tinfo->get_cwd(), m_tstr);
+			std::string sdir; // init
+			// Compare to `sinsp_filter_check_fd::extract_fdname_from_creator` logic
+			switch(evt->get_type())
+			{
+				// note: no implementation for old / legacy event definitions
+				case PPME_SYSCALL_OPENAT_2_E: // dirfd
+				case PPME_SYSCALL_OPENAT2_E:
+				{
+					int64_t dirfd = evt->get_param(0)->as<int64_t>();
+					// `parse_dirfd` checks if path is absolute and if so returns an '.' string to sdir
+					// Function tries to return fd name from the fd stored at dirfd in the fd table
+					sdir = m_inspector->get_parser()->parse_dirfd(evt, m_tstr, dirfd);
+				}
+				break;
+				case PPME_SYSCALL_OPENAT_2_X: // dirfd
+				case PPME_SYSCALL_OPENAT2_X:
+				case PPME_SYSCALL_NEWFSTATAT_X:
+				case PPME_SYSCALL_FCHOWNAT_X:
+				case PPME_SYSCALL_FCHMODAT_X:
+				case PPME_SYSCALL_MKDIRAT_X:
+				case PPME_SYSCALL_UNLINKAT_2_X:
+				case PPME_SYSCALL_MKNODAT_X:
+				{
+					int64_t dirfd = evt->get_param(1)->as<int64_t>();
+					// `parse_dirfd` checks if path is absolute and if so returns an '.' string to sdir
+					// Function tries to return fd name from the fd stored at dirfd in the fd table
+					sdir = m_inspector->get_parser()->parse_dirfd(evt, m_tstr, dirfd);
```

Comment:
Referring to the comment above, these are the calls that modify the thread state